src/main/java/com/czertainly/core/service/CertificateService.java

```diff
@@ -78,7 +78,7 @@ public interface CertificateService extends ResourceExtensionService  {
 
     CertificateContent checkAddCertificateContent(String fingerprint, String content);
 
-    CertificateDetailDto upload(UploadCertificateRequestDto request) throws AlreadyExistException, CertificateException, NoSuchAlgorithmException, NotFoundException, AttributeException;
+    CertificateDetailDto upload(UploadCertificateRequestDto request, boolean ignoreCustomAttributes) throws AlreadyExistException, CertificateException, NoSuchAlgorithmException, NotFoundException, AttributeException;
```

Comment:
Why there is `boolean ignoreCustomAttributes` in this PR? Is it related to OAuth2 implementation?

Reply:
It is part of localhost user authentication implementation which was part of this PR due to changes in authentication flow.

Ignoring custom attributes when uploading certificate is necessary when registering new local admin with new uploaded certificate. In case users are locked out of system (lost access or other reasons) don't know which custom attributes are required and that would prevent uploading new certificate to system by localhost user.